metrics/src/metrics.rs

```diff
@@ -379,4 +424,67 @@ mod test {
         agent.submit(point);
     }
 
+    #[test]
+    fn test_datapoint() {
+        macro_rules! matches {
```

Comment:
When used with `assert!`, which you always do, this is called `assert_matches!`.  https://docs.rs/assert_matches/1.3.0/assert_matches/macro.assert_matches.html